File: backend/schemas/vuln_pattern.py
# backend/schemas/vuln_pattern.py
from pydantic import BaseModel, Field
from datetime import datetime


class VulnPatternCheck(BaseModel):
    field: str
    regex: str


class VulnPatternCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=64, pattern=r'^[a-z0-9_]+$')
    description: str = Field(..., min_length=1, max_length=256)
    checks: list[VulnPatternCheck] = Field(..., min_length=1)


class VulnPatternUpdate(BaseModel):
    description: str | None = Field(None, min_length=1, max_length=256)
    checks: list[VulnPatternCheck] | None = Field(None, min_length=1)


class VulnPatternOut(BaseModel):
    id: str
    name: str
    description: str
    checks: list[VulnPatternCheck]
    is_default: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class VulnPatternTestResult(BaseModel):
    pattern_id: str
    pattern_name: str
    match_count: int
    matched_asset_ids: list[str]

File: frontend/src/types/vulnPattern.ts
// frontend/src/types/vulnPattern.ts
export interface VulnPatternCheck {
  field: string;
  regex: string;
}

export interface VulnPattern {
  id: string;
  name: string;
  description: string;
  checks: VulnPatternCheck[];
  is_default: boolean;
  created_at: string;
  updated_at: string;
}

export interface VulnPatternCreate {
  name: string;
  description: string;
  checks: VulnPatternCheck[];
}

export interface VulnPatternUpdate {
  description?: string;
  checks?: VulnPatternCheck[];
}

export interface VulnPatternTestResult {
  pattern_id: string;
  pattern_name: string;
  match_count: number;
  matched_asset_ids: string[];
}
